src/modules/rlm_pap/rlm_pap.c

```diff
@@ -96,6 +101,27 @@ static const FR_NAME_NUMBER header_names[] = {
 	{ NULL, 0 }
 };
 
+#ifdef HAVE_OPENSSL_EVP_H
+typedef struct rlm_pap_pbkdf2 {
+	char const *name;
+	EVP_MD const *(*hash_algo)(void);
+	int keylen;
+} rlm_pap_pbkdf2;
+
+static const rlm_pap_pbkdf2 pbkdf2_names[] = {
+	{ "HMACSHA1", EVP_sha1, 20 },
+	{ "HMACSHA2+224", EVP_sha224, 28 },
+	{ "HMACSHA2+256", EVP_sha256, 32 },
+	{ "HMACSHA2+384", EVP_sha384, 48 },
+	{ "HMACSHA2+512", EVP_sha512, 64 },
+//	{ "HMACSHA3+224", EVP_sha3_224, 28 },
```

Comment:
Fair enough. However the attendant Perl will generate these markers, and so I placed them as place holders (in hopes that this will soon change).